layouts/containers/TheHeader.vue

```diff
@@ -71,5 +89,31 @@ export default {
     TheHeaderDropdownAccnt,
     BreadCrumbs
   },
+  data(){
+    return {
+      languages : {
+            "ar" : "العربية",
+            "zh" : "中文",
+            "en" : "English",
+            "es" : "Español",
+            "fr" : "Françai",
+            "ru" : "Русский"
+        }
+    }
+  },
+  async fetch () { //Nuxt event to load async data at initial step 
+  },
+  mounted(){
+  },
+  methods : {
+    switchLocale(locale){
+      this.$i18n.setLocale(locale);      
+      // Get path to switch current route for selected locale
+      const switchLocalePath = this.switchLocalePath(locale);
+      this.$router.push(switchLocalePath);       
+    }
+  },
 }
 </script>
+
+<i18n src="../../i18n/dist/layouts/containers/TheHeader.json"></i18n>
```

Comment:
just a comment I would suggest to movre this `<i18n/>` above `<script>` as it will be visible and feels part of the `import` setion